### ISSUE TITLE
[SPRINT] cluster 2pt using cosmoDC2/DC2

### ISSUE THREAD

**Issue by humnaawan** [LSSTDESC/SprintWeek2020#60]
# Measuring the Cluster 2pt using cosmoDC2/DC2

Contacts: @humnaawan @cavestruz 
Day/Time: TBD
Main communication channel: #desc-cl
GitHub repo: None (yet)

## Goals and deliverable
The goal here is to measure the Cluster 2pt function using cosmoDC2/DC2 with the motivation to 1) figure out the various working parts of the analysis pipeline, and 2) validate DC2 sims.

The reach goal would be to
- Measure the Cluster 2pt function
- Figure out further investigations using the cluster 2pt, both for cluster joint analysis and joint probes analyses
- Draft a DESC Note

The stretch goal would be to
- Have a clear sense of the next project(s)
- Have a finalized DESC Note

## Resources and skills needed
Experience with DC2, interest in measuring cluster 2pt function.

## Detailed description
Please see the [project confluence page](https://confluence.slac.stanford.edu/x/YKqWEQ) for more details.

**Comment by humnaawan** [comment]
thanks @katrinheitmann @mishakb for a fruitful conversation earlier. just to recap some of the discussion points, there are a few outstanding questions before we can really measure cluster 2pt with DC2:
- what would it take for the cluster finder to be run on DC2 data? @erykoff can you please comment on this?
- @erykoff can you also say how long the cluster finding takes and whether we can expect redmapper to be runnable by anyone (e.g. aside from you)? i ask this since its likely that we will be statistics-limited with DC2 and will need SkySim5000 for our purposes.
- does CCL already have the functionality to do cluster 2pt predictions? @marina-ricci can you please comment on this (or tag someone who would know)?

@katrinheitmann please let me know if i'm missing something.

**Comment by mishakb** [comment]
@humnaawan, sure. I beleive that @pennalima may also have some comments on the CCL side.

**Comment by erykoff** [comment]
1) So I set redMaPPer running on DC2 DR6 before I took last week off, and there was some sort of problem that I am going to track down this afternoon (hopefully).
2) The calibration part was taking a few days on my wimpy old machine, it could have been faster at nersc potentially but I didn't feel like dealing with nersc (and there's also something strange that it sometimes hiccups and needs to be restarted from where it left off, and that has been more of a headache).  This calibration part only needs to be done once on a subset.  And then the time it takes scales with how many cores you throw at it.  With 12 cores for DC2 DR6 it should take about a day (but I need to fix this problem before I can confirm.)
3) It should be runnable by other people, and I know it has been run by folks in the wild (who submit feature requests and <gasp> bug reports).

**Comment by humnaawan** [comment]
thanks @erykoff! it'll be great to get progress updated on DC2 DR6 once its available.

also, i assume that the code is not set up on NERSC (please correct me if im wrong) - do you think this is something that can be achieved as a sprint during this week?

**Comment by erykoff** [comment]
It does support nersc via docker/shifter, but now I can't find my documentation on that...which would be something useful to do (properly document running at nersc!) this week after I fix the DC2 DR6 catalog generation.

**Comment by humnaawan** [comment]
@erykoff hope you're well. im just following up regarding the status of the redMaPPer output for DC2. can you please let me know if/where i can find it?
